idb/postgres/internal/writer/writer_test.go

```diff
@@ -1564,3 +1565,169 @@ func TestWriterAddBlock0(t *testing.T) {
 		assert.Equal(t, expected, accounts)
 	}
 }
+
+// Simulate a scenario where app boxes are created, mutated and deleted in consecutive rounds.
+func TestWriterAppBoxTableInsertMutateDelete(t *testing.T) {
+	/* Simulation details:
+	Box 1: inserted and then untouched
+	Box 2: inserted and mutated
+	Box 3: inserted and deleted
+	Box 4: inserted, mutated and deleted
+	Box 5: inserted, deleted and re-inserted
+	*/
+
+	db, _, shutdownFunc := pgtest.SetupPostgresWithSchema(t)
+	defer shutdownFunc()
+
+	var block bookkeeping.Block
+	block.BlockHeader.Round = basics.Round(1)
+	delta := ledgercore.StateDelta{}
+
+	blockAdder := func(tx pgx.Tx) error {
```

Comment:
nit: passing block/delta into this function would have made the test code a little clearer for me. Or maybe rename it to something like "addNewBlock", my brain read the function as "block address".